AEPMessaging/Tests/UnitTests/Cache+MessagingTests.swift

```diff
@@ -15,11 +15,272 @@ import XCTest
+        mockProposition = MessagingProposition(uniqueId: "inapp2", scope: "inapp2", scopeDetails: ["key": "value"], items: [mockPropositionItem])
+        mockSurface = Surface(uri: "inapp2")
+    }
+        
+    func getPropositionCacheEntry(_ seededPropositions: [String: [MessagingProposition]]? = nil) -> CacheEntry? {
+        let seededPropositions = seededPropositions ?? [
+            mockSurface.uri: [mockProposition]
+        ]
+        
+        let encoder = JSONEncoder()
+        guard let cacheData = try? encoder.encode(seededPropositions) else {
+            return nil
+        }
         
+        return CacheEntry(data: cacheData, expiry: .never, metadata: nil)
     }
     
+    func testPropositionsHappy() throws {
+        // setup
+        mockCache.getReturnValue = getPropositionCacheEntry()
+        
+        // test
+        let result = mockCache.propositions
+        
+        // verify
+        XCTAssertNotNil(result)
+        XCTAssertEqual(1, result?.count)
+        let propForSurface = result?[mockSurface]
+        XCTAssertNotNil(propForSurface)
+        let firstProp = propForSurface?.first
+        XCTAssertNotNil(firstProp)
+        XCTAssertEqual("inapp2", firstProp?.uniqueId)
+        XCTAssertEqual("inapp2", firstProp?.scope)
+        XCTAssertEqual("value", firstProp?.scopeDetails["key"] as? String)
+        XCTAssertEqual(1, firstProp?.items.count)
+        let propItem = firstProp?.items.first
+        XCTAssertEqual("inapp2", propItem?.itemId)
+        XCTAssertEqual(.ruleset, propItem?.schema)
+        XCTAssertNotNil(propItem?.itemData)
+        XCTAssertEqual(2, propItem?.itemData?.count)
+        XCTAssertNotNil(propItem?.itemData?["rules"])
+        XCTAssertNotNil(propItem?.itemData?["version"])
+    }
+    
+    func testPropositionsNoneInCache() throws {
+        // setup
+        mockCache.getReturnValue = nil
+        
+        // test
+        let result = mockCache.propositions
+        
+        // verify
+        XCTAssertNil(result)
+    }
+    
+    func testPropositionsCachedItemsAreNotDecodable() throws {
+        // setup
+        mockCache.getReturnValue = CacheEntry(data: "i am not valid propositions".data(using: .utf8)!, expiry: .never, metadata: nil)
+        
+        // test
+        let result = mockCache.propositions
+        
+        // verify
+        XCTAssertNil(result)
+    }
+    
+    func testUpdatePropositionsHappy() throws {
+        // setup
+        let newSurface = Surface(uri: "newSurface")
+        let newPropositionContent = JSONFileLoader.getRulesJsonFromFile("inappPropositionV2Content")
+        let newPropositionItem = MessagingPropositionItem(itemId: "inapp4", schema: .ruleset, itemData: newPropositionContent)
+        let newProposition = MessagingProposition(uniqueId: "inapp4", scope: "inapp4", scopeDetails: ["key": "value"], items: [newPropositionItem])
```

Comment:
nit: Should we match the proposition scope with the surface URI `inapp4`->`newSurface`? This is more for clarity (from managing tests perspective). A few other tests need to be updated for this.